Feat(module-ag-grid): upgrade to AG Grid 33

odinr opened the pull request:
## Why

Enhanced the AG Grid module with new functionalities and improvements.

- Added TypeScript type definitions for better type support.
- Introduced new exports for `community`, `enterprise`, and `themes`.
- Implemented `AgGridConfigurator` and `AgGridProvider` classes for better configuration and initialization.
- Added support for setting and clearing themes, managing modules, and setting license keys.
- Updated the `module.ts` to use the new configurator and provider classes.
- Removed deprecated files and refactored the module structure for better maintainability.

**Detailed Changes:**

- **package.json**: Updated dependencies and added new exports.
- **AgGridConfigurator.interface.ts**: Defined the interface for configuring AG Grid settings and modules.
- **AgGridConfigurator.ts**: Implemented the `AgGridConfigurator` class for managing AG Grid configuration.
- **AgGridProvider.ts**: Implemented the `AgGridProvider` class for providing AG Grid configuration.
- **community.ts**: Exported all from `ag-grid-community`.
- **default-modules.ts**: Added a placeholder for default modules.
- **enterprise.ts**: Exported all from `ag-grid-enterprise`.
- **index.ts**: Updated exports to use the new configurator and provider.
- **module.ts**: Refactored to use the new configurator and provider classes.
- **themes.ts**: Added a default theme for AG Grid.

**Breaking Changes:**

- `@ag-grid-community/*` and `@ag-grid-enterprise/*` must be removed from the project dependencies.
- The new configurator no longer supports setting license keys directly. Use the `setLicense` function to set license keys.
- Modules are now managed through the `AgGridConfigurator` class. Use the `setModule` or `addModule` function to set/add modules to the configuration.

**References:**

- [AG Grid 33 Migration Guide](https://www.ag-grid.com/react-data-grid/upgrading-to-ag-grid-33/)
- [AG Grid Documentation](https://www.ag-grid.com/)
- [AG Grid Theme Builder](https://www.ag-grid.com/theme-builder/)

closes:


### Check off the following:
- [x] Confirm that I checked changes to branch which I am merging into.
  - _I have validated included files_
  - _My code does not generate new linting warnings_
  - _My PR is not a duplicate, [check existing pr`s](https://github.com/equinor/fusion-framework/pulls)_
- [x] Confirm that the I have completed the [self-review checklist](https://github.com/equinor/fusion-framework/blob/main/contributing/self-review.md).

- [x] Confirm that my changes meet our [code of conduct](https://github.com/equinor/fusion-framework/blob/main/CODE_OF_CONDUCT.md).

odinr commented on packages/modules/ag-grid/src/AgGridConfigurator.ts:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    setModules(value: Module[]): void {
+        this.#modules = new Set(value);
```
see line 22

trheg-equinor commented:
Hi, some notes when trying to upgrade the fusion app Alarm Management...

I had typing issues when setting/loading modules. I added the following to my decs.d.ts:
```
declare module '@equinor/fusion-framework-module-ag-grid/community';
declare module '@equinor/fusion-framework-module-ag-grid/enterprise';

```
And then importing from either of those in my config.ts when setting modules:
```
import { ClipboardModule } from '@equinor/fusion-framework-module-ag-grid/enterprise';
import { ClientSideRowModelModule, ValidationModule } from '@equinor/fusion-framework-module-ag-grid/community';

```
Also some typing issues when trying to import interfaces from
`@equinor/fusion-framework-module-ag-grid`
I added 'dist/types/community' when importing:
```
import { ColDef, ICellRendererParams, CellClassParams } from 
'@equinor/fusion-framework-module-ag-grid/dist/types/community';
```

(Also, couldn't get the new theming method to work with setting fusionTheme in config.ts.
Using 'legacy' on the grid the v32 way works fine for me and does not break any styles)
![image](https://github.com/user-attachments/assets/dc6b7a1a-25f6-4c49-a949-da35ff290c97)

odinr commented:
> Hi, some notes when trying to upgrade the fusion app Alarm Management...
> 
> I had typing issues when setting/loading modules. I added the following to my decs.d.ts:
> 
> ```
> declare module '@equinor/fusion-framework-module-ag-grid/community';
> declare module '@equinor/fusion-framework-module-ag-grid/enterprise';
> ```
> 
> And then importing from either of those in my config.ts when setting modules:
> 
> ```
> import { ClipboardModule } from '@equinor/fusion-framework-module-ag-grid/enterprise';
> import { ClientSideRowModelModule, ValidationModule } from '@equinor/fusion-framework-module-ag-grid/community';
> ```
> 
> Also some typing issues when trying to import interfaces from `@equinor/fusion-framework-module-ag-grid` I added 'dist/types/community' when importing:
> 
> ```
> import { ColDef, ICellRendererParams, CellClassParams } from 
> '@equinor/fusion-framework-module-ag-grid/dist/types/community';
> ```
> 
> (Also, couldn't get the new theming method to work with setting fusionTheme in config.ts. Using 'legacy' on the grid the v32 way works fine for me and does not break any styles) ![image](https://private-user-images.githubusercontent.com/78970431/406859835-dc6b7a1a-25f6-4c49-a949-da35ff290c97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxNTI1ODcsIm5iZiI6MTczODE1MjI4NywicGF0aCI6Ii83ODk3MDQzMS80MDY4NTk4MzUtZGM2YjdhMWEtMjVmNi00YzQ5LWE5NDktZGEzNWZmMjkwYzk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI5VDEyMDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjNmM0NTE5MTUxODQ3NmYzNDgxOTEyZmNiOGMwNDZhNWViODg3N2RhMGZlOGU0YzZjZDY2N2U3NDZiODdmZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.C1f-JEZsFHbqbSrm6miytCPk32ZfkDYp-Qy8gv6vncY)

Hi @trheg-equinor 

Just created a fresh new application, but none of the issues that you refered to.

Could not quite understand your issues with styling.

maybe we could have a sit toghether to figure out what missing in your configuration?

package.json
```json
{
    "name": "ag-grid-33",
    "version": "0.0.0",
    "private": true,
    "type": "module",
    "main": "/src/index.ts",
    "scripts": {
        "build": "fusion-framework-cli app build",
        "dev": "fusion-framework-cli app dev"
    },
    "author": "",
    "license": "ISC",
    "dependencies": {
        "@equinor/fusion-framework-cli": "^10.4.0",
        "@equinor/fusion-framework-module-ag-grid": "33.0.0-ag-grid-v33-158362aeb14f0b9b3f798b823b0ba0e5feb3dcc1",
        "@equinor/fusion-framework-react-app": "^5.5.2",
        "ag-grid-react": "^33.0.3",
        "react": "^18.2.0",
        "react-dom": "^18.2.0",
        "typescript": "^5.5.4"
    },
    "devDependencies": {
        "@types/react": "^18.2.50",
        "@types/react-dom": "^18.2.7"
    }
}
```
tsconfig.json
```json
{
  "compilerOptions": {
    "rootDir": "src",
    "moduleResolution": "bundler",
    "module": "ES6",
    "jsx": "react-jsxdev",
  }
}
```

trheg-equinor commented:
@odinr We could, maybe tomorrow/friday.
PS. I see you are using moduleResolution "bundler". I think many fusion apps are using node / node16 / nodenext etc.
That might be the source of typing issues ( ? ) .. I can test that tomorrow

Andreas-Tonnessen commented:
Frontend tests using AG Grid v33 doesn't seem to recognize/find modules set this way:
`enableAgGrid(configurator, (builder) => {
        builder.setModules([
            ...
        ]);
    });`
So we have had to register the modules separately in the tests using `ModuleRegistry.registerModules([ ... ]);`
